### PR TITLE
Added a button in the popup to start tracking the current website

### LINKED ISSUE #11: Add a button to quickly track a new domain
If the current website is not tracked, show a button in the popup to add it to the list without going to your settings.

### PR DESCRIPTION
**Pull request by elegaanz**
Fix #11

### CODE REVIEW

File: lib/popup.html
Comment (KillianKemps) [inline lib/popup.html]: Can you reduce the space between these two `h3` elements and increase little bit the space between the text and the button?

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: I see that you are moving slowly to a MVC architecture. :D
Do you think we could use a framework like VueJS? Or is it too much for just a plugin?

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: Please add a little comment to explain that you sort the host names and only display the five first ones.

File: lib/popup.js
Comment (elegaanz) [reply lib/popup.js]: Maybe in the future, when the views will become more complex.

**Comment by KillianKemps** [comment]
Thank you for your Pull Request! I will merge it after you have changed the little things I commented.

**Comment by elegaanz** [comment]
I think it's okay now. :slightly_smiling_face:

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: I think there is a bug somewhere here.
I loaded the plugin in my browser and I see twice 'Today" in the dropdown, but no "Yesterday". Maybe is it there an issue because I have no data from day before, so it doesn't display "Yesterday"?

**Review by KillianKemps — CHANGES_REQUESTED** [review]
Thanks for the new styling of the popup, it looks better now!
I just tried the plugin in my browser and I saw a bug in the date dropdown. Could you fix it please?

File: lib/popup.js
Comment (elegaanz) [reply lib/popup.js]: I can't reproduce. If I have no data for yesterday, it simply doesn't display the 'Yesterday' `<option>`. How did you exactly get this bug (on which website, with which config, etc)?

Comment (KillianKemps) [reply lib/popup.js]: I loaded it in Firefox Developer Edition 52.0a2, and I have this issue on all websites which are tracked. For example mastodon.qowala.org (sorry, I didn't manage to take a screenshot with the dropdown open, but you have to imagine there is twice 'Today')

screenshot from 2017-05-26 13-57-42

I looked a bit more, and maybe some data is remaining from the previous version of the plugin and it does some conflict. :/

So, maybe if you load the plugin at version v0.2.1, then use it a little bit, then load the latest version would you reproduce it?

Comment (elegaanz) [reply lib/popup.js]: OK, I can reproduce if I load the add-on in my personal Firefox, not the freshly installed one `web-ext run` gives you. I can see errors in the console which makes me thinks that it is because we store (and use) DOM nodes after the popup is closed (so these DOM nodes are destroyed). :thinking:

Comment (elegaanz) [reply lib/popup.js]: It looks like the code gets executed two time, that's really strange.

capture d ecran du 2017-05-26 18 11 18

I also noticed that sometimes the popup doesn't open, and I think it is related…

Comment (KillianKemps) [reply lib/popup.js]: Oh, you have even worse issues I see :/. I have now also loaded the plugin in my normal Firefox 53.0.3, and I have again the issue in the dropdown (twice the `today` item), but I have not reproduced your bug.

Comment (elegaanz) [reply lib/popup.js]: Yes I think I created this bug when I was trying tout fix the other one. :grin:

Comment (KillianKemps) [reply lib/popup.js]: I did a `git bisect` to check where is the issue, and it seems it is the commit 0210c61967fefcf263067047491a502dc6b1874e which created the bug of having twice `Today` in the dropdown.
So we have to check the code of this commit :D

**Comment by elegaanz** [comment]
I rewrote the whole popup with Mithril. It's the first time I use it, I'm not convinced, I feel like the code is quite hard to read. But it has fixed the we had! :tada:

**Comment by KillianKemps** [comment]
I'm trying locally and it seems better, but I still see a (small) bug :smiley::
If I selected a website, for example `twitter.com`, and then change the date to "Yesterday", a date on which I did not visit this website, the popup displays "No data for this site" but the hostname is changed to `linuxfr.org`, a website for which there is in fact data for yesterday.
I didn't use Mithril either, I have only heard about. And I agree, it doesn't feel comfortable to read because the HTML is mixed up with the Javascript.
I would suggest to use Vue.js instead of Mithril, I find it nice to read. What do you think? If you think of a lighter framework it is good as well.

**Comment by elegaanz** [comment]
Well, we don't really care about size of the extension, right? So, I'll rewrite this popup for the third time with Vue. :grin:

I also have a question: what should be the behaviour when we change the date but the website that was selected was not tracked the other day (the same situation as you described)?

**Comment by KillianKemps** [comment]
Haha, if Vue.js is good for you then thank you for rewriting it a third time :D

Concerning the behavior of the popup:
Currently, when I select "Yesterday" with this state:
screenshot from 2017-05-28 21-50-46
It changes to this:
screenshot from 2017-05-28 21-50-29

This is not what I expect, because it has changed `twitter.com` to `linuxfr.org` when it has to stay the same hostname which is displayed.
With this current behavior you may think there is no Data for `linuxfr.org` for Yesterday, but in fact there is.

So, to sum up, on the second screenshot there should be `twitter.com` displayed instead of `linuxfr.org`.

Also, just thinking about it and reading your code, I think there is another bug, because the message **"No data for this site"** should have never been displayed. 
Because, if there is no data for this website, the user would be unable to select it. The only case that may happen is that there is data for a website, but not for the corresponding day, in which case it is the message **"No data for this day"** which should be displayed. If you see what I mean :D

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: Are you still working on the PR or did you forget to remove this debugging data? :smiley:

File: lib/popup.html
Comment (KillianKemps) [inline lib/popup.html]: If it is maybe better to remove this and the `noData` variable. It seems the only moment you see this, is when you change from **"Today"** to **"Yesterday"** which displays **"Please select another site"** as default value in the dropdown.
I don't see any other cases where it can be useful. So if you agree, we can remove this.

File: package.json
Comment (KillianKemps) [inline package.json]: It would be better to add Vue as a simple JS file, like you did with Mithril, instead of as a node module. Because when I use `web-ext` to build the plugin, it omits the `node_modules` folder. 
That is understandable because it contains a lot of dev dependencies which should not be packaged for production. As it is complicated to package the plugin with only the Vue.js file from the node modules without adding all other dev dependencies, it would be nice to only have a simple JS file in the `lib/` folder :)

File: manifest.json
Comment (KillianKemps) [inline manifest.json:34]: For which case did you needed to allow to evaluate strings as Javascript?

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: You can maybe move these instructions to the [`created` event hook of Vue](https://vuejs.org/v2/guide/instance.html), in order to have everything in the Vue app

File: lib/popup.js
Comment (KillianKemps) [inline lib/popup.js]: You can also move this function to a `method` of the Vue app in order to be consistent

**Comment by KillianKemps** [comment]
I like it this way with Vue.js, it feels more reactive. Thank you for the good work. :D 
Before Vue, we could see during a fraction of a second that the popup size's changed. But now it is smooth and clean.
That would be nice if you could add more unit tests, it seems easier to do now with Vue.js. But we can also already merge now, and add the tests later.
I let you take into account my last comments before we can finally merge!

File: lib/popup.js
Comment (elegaanz) [reply lib/popup.js]: Yes, I have some little adjustments to do, I'll remove that of course. :joy:

File: lib/popup.html
Comment (elegaanz) [reply lib/popup.html]: Well, it is quite disturbing to see a message like "No data on x" and nothing in the website dropdown. I think it clearly shows the user what to do to get rid of this error.

File: package.json
Comment (elegaanz) [reply package.json]: OK, no problem with that.

File: manifest.json
Comment (elegaanz) [reply manifest.json:34]: Vue use it all the time.

File: lib/popup.js
Comment (elegaanz) [reply lib/popup.js]: Yes, why not.

File: lib/popup.html
Comment (KillianKemps) [reply lib/popup.html]: Yes it is good to be explicit and inform the user, but here I don't understand the use case. 
Looking at your code, the message `You didn't visited this site ${this.date.toLowerCase()}!` only displays when the website dropdown is on **"Please select another site"**. It seem it may not happen in any other case.

Trying the plugin locally, when you switch the date dropdown, by default it goes to **"Please select another site"**, so you have the message **"You didn't visited this site yesterday!"** which is displayed. But if you would remove this from the dropdown, you will never have the case that data is missing for a selected website, because it is impossible for you to select a website for which there is no data, as it is not displayed in the dropdown.
I really don't understand in which case this is useful ^^'

What is the case you encountered which made you add this logic?

**Comment by elegaanz** [comment]
I think everything is okay now. I hope I didn't forget to fix/do something. :grin:

**Review by KillianKemps — APPROVED** [review]

**Comment by KillianKemps** [comment]
Seems good, thank you!
I will merge it now, and we can add more tests later.

**Pull request merged**
